src/media/js/payments.js

```diff
@@ -1,129 +1,158 @@
+            fxpaySettings: null,
+        });
+        var purchase = defer.Deferred();
 
         if (!product || !product.payment_required) {
-            return $def.resolve(product).promise();
+            return purchase.resolve(product).promise();
         }
 
-        console.log('Initiating transaction');
+        logger.log('Initiating transaction');
 
-        if (caps.navPay || settings.simulate_nav_pay) {
-            requests.post(urls.api.url('prepare_nav_pay'), {app: product.slug}).done(function(result) {
-                console.log('Calling mozPay with JWT: ', result.webpayJWT);
-                var request;
-                if (caps.navPay && !settings.simulate_nav_pay) {
-                    request = navigator.mozPay([result.webpayJWT]);
-                } else {
-                    request = navigator.fakeMozPay([result.webpayJWT]);
+        if (opt.fxpaySettings) {
+            fxpay.configure(opt.fxpaySettings);
+        }
+
+        fxpay.purchase(product.slug, function(error) {
+            if (error) {
+                var msg;
+                logger.error('fxpay error:', error);
+                if (error === 'APP_ALREADY_PURCHASED') {
+                    return purchase.resolve(product);
                 }
-                request.onsuccess = function() {
-                    console.log('navigator.mozPay success');
-                    waitForPayment($def, product, result.webpayJWT, result.contribStatusURL);
-                };
-                request.onerror = function(errorMsg) {
-                    var msg;
-                    console.error('`navigator.mozPay` error:', this.error.name);
-                    switch (this.error.name) {
-                        // Sent from webpay.
-                        case 'USER_CANCELLED':
-                        // Sent from the trusted-ui on cancellation.
-                        case 'DIALOG_CLOSED_BY_USER':
-                            msg = gettext('Payment cancelled.');
-                            break;
-                        default:
-                            msg = gettext('Payment failed. Try again later.');
-                            break;
-                    }
-
-                    notify({
-                        classes: 'error',
-                        message: msg,
-                        timeout: 5000
-                    });
-
-                    $def.reject(null, product, 'MKT_CANCELLED');
-                };
-            }).fail(function(xhr, status, error) {
-                if (error === 409) {
-                    console.warn('App already purchased (409 from API)');
-                    // This error code means the user has already purchased the app.
-                    $def.resolve(product);
-                    return;
+                switch (error) {
+                    // Sent from webpay.
+                    case 'USER_CANCELLED':
+                    // Sent from the trusted-ui on cancellation.
```

Comment:
acutally, I can't find the source of this string anywhere! I can't find it in Spartacus, can't find it in DXR (moz central)